tensorflow/python/ops/variables.py

```diff
@@ -226,7 +226,11 @@ def __init__(self,
           expected_shape=expected_shape)
 
   def __str__(self):
-    return str(self._snapshot)
+    return self.__repr_()
```

Comment:
You actually don't need a `__str__` method at all if it should be the same as  `__repr__`.